### ISSUE TITLE
Error exporting

### ISSUE THREAD

**Issue by fedtpelsen** [retorquere/zotero-better-bibtex#576]
I have Zotero standalone, chrome ext and BBT installed. I can export BibTeX and BibLaTeX fine but exporting the Better-versions give me a javascript error. 

Here's the ID: EAFUV3B3.

**Comment by retorquere** [comment]
@fedtpelsen, @stfl: could you please enable "extended debugging" in the BBT preferences, try another export, and submit a new error report? The current logs have only the error, but not the activity leading up to the error.

**Comment by stfl** [comment]
report id: P3SCK9RW

**Comment by fedtpelsen** [comment]
New ID with extended debugging: WS9ZJDIB

**Comment by retorquere** [comment]
Same problem in both cases. The translator is being handed an empty object, but I can't replicate that on my end. Can you both try https://github.com/retorquere/zotero-better-bibtex/releases/download/builds/zotero-better-bibtex-1.6.77-circle-3110.xpi and subit a new debug log?

**Comment by stfl** [comment]
UHGR3G3A

**Comment by retorquere** [comment]
This is strange -- I'm getting reference objects that haven't passed through my pipeline. Can you try again with https://github.com/retorquere/zotero-better-bibtex/releases/download/builds/zotero-better-bibtex-1.6.77-circle-3113.xpi ?

I can't yet reproduce this on my end alas, or I wouldn't be bothering you.

**Comment by stfl** [comment]
ID37J4A4

**Comment by stfl** [comment]
I just downgraded to 1.6.75 and it works..

**Comment by retorquere** [comment]
Dammit, that rules out one possible cause. You're on Linux, right? Could you get me a [real-time debug log](https://www.zotero.org/support/debug_output)? They say to copy to the clipboard, but just start `script` before you start Zotero. After you've closed Zotero, if you type `exit`, the whole log will be in a file called `typescript`.

**Comment by retorquere** [comment]
[this version](https://github.com/retorquere/zotero-better-bibtex/releases/download/builds/zotero-better-bibtex-1.6.77-circle-3114.xpi) may also tell me more, but I'm afraid BBT is erroring out before I even get to set up the pipeline.

**Comment by retorquere** [comment]
While a real-time log will have the most information, [this](https://github.com/retorquere/zotero-better-bibtex/releases/download/builds/zotero-better-bibtex-1.6.77-circle-3115.xpi) version has been instrumented to disable itself if it encounters an initialization error (which almost certainly must be the case) and will display that error in the BBT preferences. It may have enough information, and it'll be easier for you (the realtime debug log is not really an easy process).

**Comment by stfl** [comment]
I will give it a go in a minute

**Comment by retorquere** [comment]
60... 59... 58...

(jk, I'm starting my commute right now so I won't be able to do anything for the next 30 minutes or so)

If it is indeed an init error:

1. It'll probably be fairly easy to fix, and
2. I have no idea why I'm not experiencing it.

**Comment by stfl** [comment]
when I try to open the preferences: 

```
Initialize failed: TypeError: Zotero.BetterBibTeX.CSL.DateParser.addDateParserMonths is not a function
```

**Comment by stfl** [comment]
with version 3115:
[typescript.txt](https://github.com/retorquere/zotero-better-bibtex/files/536695/typescript.txt)

[3113.txt](https://github.com/retorquere/zotero-better-bibtex/files/536709/3113.txt)

**Comment by retorquere** [comment]
Well that localizes the issue but doesn't explain why you're getting an error and I'm not. So frustrating.

Could you try the same (a realtime debug log) with both [this version](https://github.com/retorquere/zotero-better-bibtex/releases/download/builds/zotero-better-bibtex-1.6.77-circle-3116.xpi) (will still fail but hopefully with informative logging) and [this version](https://github.com/retorquere/zotero-better-bibtex/releases/download/builds/zotero-better-bibtex-1.6.77-circle-3117.xpi) (there is a decent chance this version will just work, but I'd be a little disappointed if this were the problem).

I think I've isolated the change between .77 and .75 that could yield this particular error; .75 bundled a private copy of the Zotero citation processor, .77 uses the citation processor that's already built into Zotero, but that doesn't explain why there is apparently a difference between the bundled processor you have and the one I have.

**Comment by stfl** [comment]
Could it be related to the underlying firefox version? 
I am using the firefox-esr from debian stable (don't know the version now)

And i haven't tried version 1.6.76

**Comment by retorquere** [comment]
Should really more have to do with the Zotero version rather than the Firefox version. If I have identified the problem correctly, .76 should have the same issue as .77. The two versions I linked to above (variants on .77) will tell me more.

That profile picture of yours, is that from a Holi celebration?

**Comment by stfl** [comment]
Yes. From holi in Kathmandu :D
I semd you the report tomorrow

**Comment by retorquere** [comment]
Looking forward to that report, keen to get this solved. I haven't been able to reproduce on my end.

Where I was, Holi got out of hand really rapidly. Colors, yes, but also egg and waste motor oil.

**Comment by stfl** [comment]
[3116.txt](https://github.com/retorquere/zotero-better-bibtex/files/538289/3116.txt)

3117 works..
[3117.txt](https://github.com/retorquere/zotero-better-bibtex/files/538290/3117.txt)

**Comment by fedtpelsen** [comment]
I still get an error with .77

V6PRRVM2

**Comment by fedtpelsen** [comment]
3177 works for me too!

**Comment by fedtpelsen** [comment]
3117 ofc.

**Comment by retorquere** [comment]
Thanks for the confirmation, tests are running. A little disappointing, but so it goes.

**Comment by retorquere** [comment]
This has been merged to master; when #579 clears I'll cut a new release.

**Comment by retorquere** [comment]
new release is out.
